ui/app/hooks/useTimeout.js

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect, useRef, useCallback } from 'react'
+
+/**
+ * useTimeout
+ *
+ * @param {Function}  cb   - callback function inside setTimeout
+ * @param {number}  delay   - delay in ms
+ * @param {boolean}  immediate   - delay in ms
+ *
+ * @return {Function}
+ */
+export function useTimeout (cb, delay, immediate = true) {
+  const saveCb = useRef()
+  const [triggered, setTriggered] = useState(immediate)
+
+  useEffect(() => {
+    saveCb.current = cb
+  }, [cb])
+
+  useEffect(() => {
+    if (!triggered) {
+      return
+    }
+
+    const id = setTimeout(() => {
+      setTriggered(false)
+      saveCb.current()
+    }, delay)
+
+    return () => {
+      clearTimeout(id)
+    }
+  }, [delay, triggered])
+
+  const trigger = useCallback(() => {
```

Comment:
Nit: I was a bit confused by this name at first; could this be changed to something like `startTimeout`?